File: src/main/resources/static/ts/jsuis/JSProperties.ts
/// <reference path = "../jsuis.ts"/>
/**
 * JSProperties
 * 
 * @author Yassuo Toda
 */
class JSProperties {

    properties: { [ key: string ]: string };
    
    constructor() {
        this.setProperties({});
    }
    
    getProperties() {
        return this.properties;
    }
    setProperties(properties: { [ key: string ]: string }) {
        this.properties = properties;
    }
    getProperty(key: string): string;
    getProperty(key: string, defaultValue: string): string;
    // overload
    getProperty(...args: any[]): string {
        var value: string;
        var properties = this.getProperties();
        switch (args.length) {
        case 1:
            // getProperty(key: string): string;
            if (typeof args[0] === "string") {
                var key: string = args[0];
                value = properties[key];
            }
        case 2:
            // getProperty(key: string, defaultValue: string): string;
            if (typeof args[0] === "string" && typeof args[1] === "string") {
                var key: string = args[0];
                var defaultValue: string = args[1];
                value = properties[key];
                if (value === undefined) {
                    value = defaultValue;
                }
            }
            break;
        default:
        }
        return value;
    }
    setProperty(key: string, value: string) {
        var properties = this.getProperties();
        properties[key] = value;
    }
}

File: jsuis-model/src/main/java/jsuis/util/JSProperties.java
package jsuis.util;

import java.io.IOException;
import java.io.Reader;
import java.io.Writer;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Properties;

/**
 * Properties
 * 
 * @author Yassuo Toda
 */
public class JSProperties extends Properties {

	private static final long serialVersionUID = 1L;

	private Path path;
	
	public JSProperties(Path path) throws IOException {
		this.path = path;
		load();
	}
	
	public JSProperties set(String key, String value) {
		setProperty(key, value);
		return this;
	}
	
	public void load() throws IOException {
		if (Files.exists(path)) {
			try (Reader reader = Files.newBufferedReader(path, StandardCharsets.ISO_8859_1)) {
				load(reader);
			}
		}
	}
	
	public void store() throws IOException {
		try (Writer writer = Files.newBufferedWriter(path, StandardCharsets.ISO_8859_1)) {
			store(writer, null);
		}
	}
}
